### PR TITLE
db48: add support for MACOSX_DEPLOYMENT_TARGET > 10

### PR DESCRIPTION
**Pull request by MarcusCalhoun-Lopez**
###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.2.3 20D91
Xcode 12.4 12D4e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: databases/db48/files/yosemite-libtool.patch [hunk 1]
```diff
@@ -2,12 +2,15 @@ Index: dist/configure
...
+@@ -8840,9 +8840,9 @@
        case ${MACOSX_DEPLOYMENT_TARGET-10.0},$host in
  	10.0,*86*-darwin8*|10.0,*-darwin[91]*)
  	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
 -	10.[012]*)
 +	10.[012][,.]*)
  	  _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
- 	10.*)
+-	10.*)
++	10.*|11.*)
```
Comment (jmroot) [inline databases/db48/files/yosemite-libtool.patch:13]: Should probably allow any initial number >= 10, after all macOS 12 isn't that far away.

**Review by jmroot — COMMENTED** [review]
Given that this currently builds fine on Big Sur, I suspect it doesn't actually need to allow undefined symbols anywhere. But I suppose this doesn't hurt.

File: databases/db48/files/yosemite-libtool.patch (hunk 1 above)
Comment (MarcusCalhoun-Lopez) [reply databases/db48/files/yosemite-libtool.patch:13]: I should have mentioned that it only causes an error with `db48 +tcl`.

Comment (MarcusCalhoun-Lopez) [reply databases/db48/files/yosemite-libtool.patch]: I changed the patch to be more in line with [Ticket 61488](https://trac.macports.org/ticket/61488).

Comment (jmroot) [reply databases/db48/files/yosemite-libtool.patch]: > I should have mentioned that it only causes an error with `db48 +tcl`.

Ah, good to know. Thanks.

**Pull request merged**